src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/MarkJavaObjects.cs

```diff
@@ -149,13 +149,55 @@ void PreserveAdapter (TypeDefinition type)
 			}
 		}
 
+		string TypeNameWithoutKey (string name)
+		{
+			var idx = name.IndexOf (", PublicKeyToken=");
+			if (idx > 0)
+				name = name.Substring (0, idx);
+
+			return name;
+		}
+
+		bool CheckInvokerType (TypeDefinition type, string name)
+		{
+			return TypeNameWithoutKey (name) == TypeNameWithoutKey ($"{ type.FullName}, { type.Module.Assembly.FullName}");
+		}
+
+		void PreserveInterfaceMethods (TypeDefinition type, TypeDefinition invoker)
+		{
+			foreach (var m in type.GetMethods ()) {
+				CustomAttribute registerAttr;
+				if (!m.TryGetRegisterAttribute (out registerAttr))
+					continue;
+
+				if (registerAttr.ConstructorArguments.Count != 3)
+					continue;
+
+				var methodAndType = (string)registerAttr.ConstructorArguments [2].Value;
+				if (!methodAndType.Contains (":"))
+					continue;
+
+				var values = methodAndType.Split (':');
```

Comment:
This should use `string.Split(char[], int)`, e.g.

```
var values = methodAndType.Split (new[]{':'}, 2);
```

That way if the type or assembly name *somehow* contains `:`, we don't screw up.